lib/ruby_lsp/requests/request.rb

```diff
@@ -65,6 +65,22 @@ def determine_target(target, parent, position)
 
         target
       end
+
+      # Checks if a given location covers the position requested
+      sig { params(location: T.nilable(Prism::Location), position: T::Hash[Symbol, T.untyped]).returns(T::Boolean) }
+      def covers_position?(location, position)
+        return false unless location
+
+        start_line = location.start_line - 1
+        end_line = location.end_line - 1
+
+        start_covered = start_line < position[:line] ||
+          (start_line == position[:line] && location.start_column <= position[:character])
+        end_covered = end_line > position[:line] ||
+          (end_line == position[:line] && location.end_column >= position[:character])
+
+        start_covered && end_covered
```

Comment:
I know this makes the code more readable, but we are missing an optimization here that we can get with an early return if we inline both branches.

In other words, if we fail `start_covered` check, we shouldn't have to make extra work to calculate `end_covered`.

Also, it would be good to access `position` fields once and not multiple times.

Reply:
```suggestion
        position_line = position[:line]
        position_char = position[:character]
        
        (start_line < position_line ||
          (start_line == position_line && location.start_column <= position_char)) &&
        (end_line > position_line ||
          (end_line == position_line && location.end_column >= position_char))
```

Reply:
Good points. I got rid of the `start_covered` and `end_covered` variables, so that the conditional short circuits when failing.

Also, putting the position parts into variables made it more readable, so we're not missing out on anything.